x-pack/packages/kbn-elastic-assistant/impl/assistant/index.tsx

```diff
@@ -234,13 +241,29 @@ const AssistantComponent: React.FC<Props> = ({
   ////
   //
 
-  const selectedSystemPrompt = useMemo(() => {
-    if (currentConversation.apiConfig.defaultSystemPromptId) {
-      return allSystemPrompts.find(
-        (prompt) => prompt.id === currentConversation.apiConfig.defaultSystemPromptId
+  const selectedSystemPrompt = useMemo(
+    () => getDefaultSystemPrompt({ allSystemPrompts, conversation: currentConversation }),
+    [allSystemPrompts, currentConversation]
+  );
+
+  const [editingSystemPromptId, setEditingSystemPromptId] = useState<string | undefined>(
```

Comment:
This is the new intermediate state for the currently-edited system prompt. `undefined` represents the "no selection" state.